src/Tools/Source/RunTests/ProcessTestExecutor.cs

```diff
@@ -91,7 +101,10 @@ private string GetResultsFilePath(AssemblyInfo assemblyInfo)
                     lowPriority: false,
                     displayWindow: false,
                     captureOutput: true,
-                    cancellationToken: cancellationToken);
+                    cancellationToken: cancellationToken,
+                    environmentVariables: environmentVariables,
+                    onProcessStartHandler: (process) => ProcDumpRunner.StartProcDump(process.Id, process.ProcessName, outputDirectory, Logger.Log)
```

Comment:
As I see, in any case, we can have a race condition here. We could double-check that the target process exists just before launching procdump. However, the process can terminate between the check and the run of procdump. 
My understanding that the outcome of the race condition would be a void procdump process running during the test session. I think that the test machine will be cleaned between sessions and this will terminate the procdump. As an alternate option, we can keep all process ids of procdump processes and try to terminate them at the end of the session. Would it be better?

---
In reply to: [122554027](https://github.com/dotnet/roslyn/pull/20129#discussion_r122554027) [](ancestors = 122554027)